akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala

```diff
@@ -82,6 +82,7 @@ private[headers] sealed trait SyntheticHeader extends ModeledHeader
  * as they allow the custom header to be matched from [[RawHeader]] and vice-versa.
  */
 abstract class CustomHeader extends jm.headers.CustomHeader {
+  HttpHeader.validateHeaderValue(value)
```

Comment:
Hmm, this one cannot work because of initialization issues. I.e. if the custom implementation of `def value(): String` depends on a custom field that will not yet have been initialized when the validation method is called. Maybe we need to move it to `render` in that case?

Reply:
@jrudolph, How about moving it to [apply](https://github.com/akka/akka-http/pull/3763/files#diff-f6cd1e0396f76b9123778a986d9c55a3b43a69c27c949320c30456a2f4a61e29R100) method?